### PR TITLE
OPS-1433 grant-ssh access job should fail if user has no keys

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops kindly review. Thanks

[Job Reference](ansible-grant-ssh-access/1244/console)

### CODE REVIEW

File: playbooks/roles/user/tasks/main.yml [hunk 1]
```diff
@@ -98,29 +98,40 @@
...
+- name: Check the ssh key(s) for user(s) over github
+  uri:
+    url: "https://github.com/{{ item.name }}.keys"
+  with_items: "{{ user_info }}"
+  register: github_users_return
+
+- fail:
+    msg: "User {{ item.item.name }} didn't added ssh key to his account over github"
+  with_items: "{{ github_users_return.results }}"
+  when: item.content_length == "0"
```
Comment (e0d) [inline playbooks/roles/user/tasks/main.yml:125]: Is this a reliable test?  Out of curiosity do we get a 404 response?

**Comment by e0d** [comment]
:+1:

File: playbooks/roles/user/tasks/main.yml (hunk 1 above)
Comment (arbabnazar) [inline playbooks/roles/user/tasks/main.yml:125]: yes, it's reliable test. didn't get 404 error. Thanks

File: playbooks/roles/user/tasks/main.yml [hunk 2]
```diff
@@ -98,29 +98,40 @@
...
 
-- name: get github key(s) and update the authorized_keys file
+- name: Check the ssh key(s) for user(s) over github
+  uri:
+    url: "https://github.com/{{ item.name }}.keys"
+  with_items: "{{ user_info }}"
+  register: github_users_return
+
+- fail:
+    msg: "User {{ item.item.name }} didn't added ssh key to his account over github"
```
Comment (feanil) [inline playbooks/roles/user/tasks/main.yml:123]: A more clear way of saying this might be: `User {{ item.item.name }} doesn't have an SSH key associated with their account`

Comment (arbabnazar) [inline playbooks/roles/user/tasks/main.yml:123]: Thanks @feanil, I'll modified it in new PR

**Comment by jibsheet** [comment]
@arbabnazar this breaks AMI builds because there are old absent users in our config without public ssh keys.  Ideally it wouldn't die on absent users without ssh keys.

**Comment by arbabnazar** [comment]
@jibsheet Thanks for pointing and addressing it into separate PR.

**Comment by jibsheet** [comment]
@arbabnazar this additionally causes failures documented in https://openedx.atlassian.net/browse/DEVOPS-4400

TL;DR - original version of this used fetch_url which uses urllib which is installed in 12.04 and 14.04
Updated version using uri: uses httplib2 which is installed on 12.04 but is not on 14.04.

Fixing in abbey.py so that we can build IDAs deployed on 14.04

**Comment by jibsheet** [comment]
Further fallout https://github.com/edx/configuration/pull/3177
